### PR TITLE
post-date-integration-support for qbo changes

### PR DESCRIPTION
**Pull request by DhaaraniCIT**

### CODE REVIEW

**Review by ashwin1111 — COMMENTED** [review]
let's accommodate the change discussed here https://fylein.slack.com/archives/C05BPQDGWUD/p1687446511632019?thread_ts=1687178637.598439&cid=C05BPQDGWUD

File: src/app/shared/components/configuration/export-settings/export-settings.component.ts [hunk 1]
```diff
@@ -93,7 +93,7 @@ export class ExportSettingsComponent implements OnInit, OnDestroy {
     }
   ];
 
-  cccExpenseGroupingDateOptions: ExportSettingFormOption[] = this.reimbursableExpenseGroupingDateOptions.concat();
+  cccExpenseGroupingDateOptions: ExportSettingFormOption[] = this.reimbursableExpenseGroupingDateOptions;
```
Comment (ashwin1111) [inline src/app/shared/components/configuration/export-settings/export-settings.component.ts:96]: we should do .concat, why are we removing them?

File: src/app/shared/components/configuration/export-settings/export-settings.component.ts [hunk 2]
```diff
@@ -244,11 +244,28 @@ export class ExportSettingsComponent implements OnInit, OnDestroy {
...
+  }
+
+  cccExpenseGroupFn() {
+    if (this.exportSettingsForm.controls.creditCardExportGroup.value === ExpenseGroupingFieldOption.EXPENSE_ID) {
+      this.cccExpenseGroupingDateOptions = this.reimbursableExpenseGroupingDateOptions.concat([{
+        label: 'Posted Date',
+        value: ExportDateType.POSTED_AT
+      }]);
+    } else {
+      this.cccExpenseGroupingDateOptions = this.reimbursableExpenseGroupingDateOptions;
```
Comment (ashwin1111) [inline src/app/shared/components/configuration/export-settings/export-settings.component.ts:268]: .concat()

File: src/app/shared/components/configuration/export-settings/export-settings.component.ts [hunk 3]
```diff
@@ -244,11 +244,28 @@ export class ExportSettingsComponent implements OnInit, OnDestroy {
...
+          value: ExportDateType.SPENT_AT
+        }];
     } else {
       this.exportSettingsForm.controls.creditCardExportGroup.enable();
-      this.exportSettingsForm.controls.creditCardExportDate.enable();
+      this.cccExpenseGroupFn();
+    }
+  }
+
+  cccExpenseGroupFn() {
```
Comment (ashwin1111) [inline src/app/shared/components/configuration/export-settings/export-settings.component.ts:261]: let's expect creditCardExportGroup value here. If you call this function from the valuechanges watcher, ideally the correct form value would've not been set

File: src/app/shared/components/configuration/export-settings/export-settings.component.ts (hunk 1 above)
Comment (DhaaraniCIT) [reply src/app/shared/components/configuration/export-settings/export-settings.component.ts:96]: by mistake it was removed

File: src/app/shared/components/configuration/export-settings/export-settings.component.ts [hunk 4]
```diff
@@ -244,11 +244,28 @@ export class ExportSettingsComponent implements OnInit, OnDestroy {
...
+          value: ExportDateType.SPENT_AT
+        }];
     } else {
       this.exportSettingsForm.controls.creditCardExportGroup.enable();
-      this.exportSettingsForm.controls.creditCardExportDate.enable();
+      this.cccExpenseGroupFn(this.exportSettingsForm.controls.creditCardExportGroup.value);
+    }
+  }
+
+  cccExpenseGroupFn(creditCardExportGroup: ExpenseGroupingFieldOption) {
```
Comment (ashwin1111) [inline src/app/shared/components/configuration/export-settings/export-settings.component.ts:261]: add return type, rename function

File: src/app/shared/components/configuration/export-settings/export-settings.component.ts [hunk 5]
```diff
@@ -254,11 +254,11 @@ export class ExportSettingsComponent implements OnInit, OnDestroy {
         }];
     } else {
       this.exportSettingsForm.controls.creditCardExportGroup.enable();
-      this.cccExpenseGroupFn(this.exportSettingsForm.controls.creditCardExportGroup.value);
+      this.getCreditCardExpenseGroupingDateOptions(this.exportSettingsForm.controls.creditCardExportGroup.value);
     }
   }
 
-  cccExpenseGroupFn(creditCardExportGroup: ExpenseGroupingFieldOption) {
+  getCreditCardExpenseGroupingDateOptions(creditCardExportGroup: ExpenseGroupingFieldOption): void {
```
Comment (ashwin1111) [inline src/app/shared/components/configuration/export-settings/export-settings.component.ts:261]: instead of get, call it set

**Review by ashwin1111 — APPROVED** [review]

**Pull request merged**